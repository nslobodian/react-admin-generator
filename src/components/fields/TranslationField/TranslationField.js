import React, { Component } from 'react'
import Tabs, { TabPane } from 'rc-tabs'
import TabContent from 'rc-tabs/lib/TabContent'
import TabBar from 'rc-tabs/lib/TabBar'
import Field from 'redux-form/es/Field'
import FormSection from 'redux-form/es/FormSection'
import 'rc-tabs/assets/index.css'
import s from './TranslationField.scss'
import connect from 'react-redux/es/connect/connect'
import { getLanguagesAttempt, getCountriesAttempt } from 'redux/Languages'

export const ChangeTabField = ({ input: { onChange, value }, children, countries, needCode, innerWrap, forceRender }) => (
  <Tabs
    activeKey={value || 'sk'}
    onChange={onChange}
    renderTabBar={() => <TabBar />}
    renderTabContent={() => <TabContent />}
  >
    {
      countries.map(({ language: { code, required }, currency, vat }, index) =>
        <TabPane tab={`${code.toUpperCase()} ${required ? '*' : ''}`} key={code} forceRender={forceRender}>
          {
            innerWrap
              ? needCode ? React.cloneElement(children, { code, currencyCode: currency.code, vat, conversionRate: currency.conversionRate }) : children
              : <FormSection name={code} required={required}>
                {needCode ? React.cloneElement(children, { code, currencyCode: currency.code, vat, conversionRate: currency.conversionRate }) : children}
              </FormSection>
          }
        </TabPane>
      )
    }
  </Tabs>

)

export class TranslationField extends Component {
  componentWillMount () {
    this.props.getCountriesAttempt()
  }

  handleChange = (e, value) => {
    e.preventDefault()
    this.props.handleChangeTab(value)
  }

  render () {
    const { countries, name, children, needCode = true, innerWrap, tabKey = 'tabKey', forceRender, handleChangeTab } = this.props
    const FieldComponet = <Field component={ChangeTabField} name={tabKey} countries={countries} children={children}
      needCode={needCode} innerWrap={innerWrap} forceRender={forceRender} onChange={handleChangeTab ? this.handleChange : null} />
    return <div className={s.TranslationField}>
      {name
        ? <FormSection name={name}>{FieldComponet}</FormSection>
        : FieldComponet
      }
    </div>
  }
}

const mapActionToProps = {
  getLanguagesAttempt,
  getCountriesAttempt,
}
const mapStateToProps = ({ Languages: { countries } }) => ({ countries })

export default connect(mapStateToProps, mapActionToProps)(TranslationField)
