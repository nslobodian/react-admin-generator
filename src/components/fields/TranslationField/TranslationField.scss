.TranslationField {
  :global {
    .rc-tabs-bar {
      padding: 5px;

      .rc-tabs-tab {
        margin-right: 5px;
      }
    }
  }
}
