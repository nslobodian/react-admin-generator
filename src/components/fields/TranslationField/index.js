import TranslationField from './TranslationField'

export default TranslationField
