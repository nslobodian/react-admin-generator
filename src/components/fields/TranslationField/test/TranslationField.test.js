import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { ChangeTabField, TranslationField } from '../TranslationField'

describe('(Components) TranslationField', () => {
  it('Should render component properly', () => {
    const props = {
      children: [],
      input: {
        onChange: jest.fn(),
        value: 'sk',
      },
      countries: [
        {
          inde: 0,
          language: {
            code: 'abc',
            required: true,
          },
          currency: 'EUR',
        },
      ],
      needCode: true,
      innerWrap: true,
    }
    const wrapper = shallow(
      <ChangeTabField
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('Should render component properly without some attribues', () => {
    const props = {
      children: [],
      input: {
        onChange: jest.fn(),
      },
      countries: [
        {
          inde: 0,
          language: {
            code: 'abc',
            required: false,
          },
          currency: 'EUR',
        },
      ],
      needCode: false,
      innerWrap: false,
    }
    const wrapper = shallow(
      <ChangeTabField
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('Should render component properly without some attribues more', () => {
    const props = {
      children: [],
      input: {
        onChange: jest.fn(),
      },
      countries: [
        {
          inde: 0,
          language: {
            code: 'abc',
            required: false,
          },
          currency: 'EUR',
        },
      ],
      needCode: true,
      innerWrap: false,
    }
    const wrapper = shallow(
      <ChangeTabField
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('Should render component properly without some attribues more more', () => {
    const props = {
      children: [],
      input: {
        onChange: jest.fn(),
      },
      countries: [
        {
          inde: 0,
          language: {
            code: 'abc',
            required: false,
          },
          currency: 'EUR',
        },
      ],
      needCode: false,
      innerWrap: true,
    }
    const wrapper = shallow(
      <ChangeTabField
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('Should render TranslationField properly', () => {
    const props = {
      children: [],
      name: 'name',
      tabKey: '',
      countries: [
        {
          inde: 0,
          language: {
            code: 'abc',
            required: false,
          },
          currency: 'EUR',
        },
      ],
      needCode: false,
      innerWrap: true,
      getCountriesAttempt: jest.fn(),
    }
    const wrapper = shallow(
      <TranslationField
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  }),
  it('Should render TranslationField properly without some attributes', () => {
    const props = {
      children: [],
      countries: [
        {
          inde: 0,
          language: {
            code: 'abc',
            required: false,
          },
          currency: 'EUR',
        },
      ],
      innerWrap: true,
      getCountriesAttempt: jest.fn(),
    }
    const wrapper = shallow(
      <TranslationField
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
