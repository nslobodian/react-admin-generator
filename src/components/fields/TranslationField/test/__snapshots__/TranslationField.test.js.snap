// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Components) TranslationField Should render TranslationField properly 1`] = `
<div
  className="TranslationField"
>
  <FormSection
    component="div"
    name="name"
  >
    <Field
      component={[Function]}
      countries={
        Array [
          Object {
            "currency": "EUR",
            "inde": 0,
            "language": Object {
              "code": "abc",
              "required": false,
            },
          },
        ]
      }
      innerWrap={true}
      name=""
      needCode={false}
    />
  </FormSection>
</div>
`;

exports[`(Components) TranslationField Should render TranslationField properly without some attributes 1`] = `
<div
  className="TranslationField"
>
  <Field
    component={[Function]}
    countries={
      Array [
        Object {
          "currency": "EUR",
          "inde": 0,
          "language": Object {
            "code": "abc",
            "required": false,
          },
        },
      ]
    }
    innerWrap={true}
    name="tabKey"
    needCode={true}
  />
</div>
`;

exports[`(Components) TranslationField Should render component properly 1`] = `
<Tabs
  activeKey="sk"
  destroyInactiveTabPane={false}
  onChange={[Function]}
  prefixCls="rc-tabs"
  renderTabBar={[Function]}
  renderTabContent={[Function]}
  style={Object {}}
  tabBarPosition="top"
>
  <TabPane
    placeholder={null}
    tab="ABC *"
  >
    <undefined
      code="abc"
    />
  </TabPane>
</Tabs>
`;

exports[`(Components) TranslationField Should render component properly without some attribues 1`] = `
<Tabs
  activeKey="sk"
  destroyInactiveTabPane={false}
  onChange={[Function]}
  prefixCls="rc-tabs"
  renderTabBar={[Function]}
  renderTabContent={[Function]}
  style={Object {}}
  tabBarPosition="top"
>
  <TabPane
    placeholder={null}
    tab="ABC "
  >
    <FormSection
      component="div"
      name="abc"
      required={false}
    />
  </TabPane>
</Tabs>
`;

exports[`(Components) TranslationField Should render component properly without some attribues more 1`] = `
<Tabs
  activeKey="sk"
  destroyInactiveTabPane={false}
  onChange={[Function]}
  prefixCls="rc-tabs"
  renderTabBar={[Function]}
  renderTabContent={[Function]}
  style={Object {}}
  tabBarPosition="top"
>
  <TabPane
    placeholder={null}
    tab="ABC "
  >
    <FormSection
      component="div"
      name="abc"
      required={false}
    >
      <undefined
        code="abc"
      />
    </FormSection>
  </TabPane>
</Tabs>
`;

exports[`(Components) TranslationField Should render component properly without some attribues more more 1`] = `
<Tabs
  activeKey="sk"
  destroyInactiveTabPane={false}
  onChange={[Function]}
  prefixCls="rc-tabs"
  renderTabBar={[Function]}
  renderTabContent={[Function]}
  style={Object {}}
  tabBarPosition="top"
>
  <TabPane
    placeholder={null}
    tab="ABC "
  />
</Tabs>
`;
